### PR TITLE
Setting index name / names for Series

### PR DESCRIPTION
**Pull request by itholic**
Unlike pandas, `koalas.Series` can't set the index name like below.

```python
>>> pser = pd.Series([1, 2, 3, 4, 5])
>>> pser.index.name = 'koalas'
>>> pser.index.name
'koalas'

>>> kser = ks.Series([1, 2, 3, 4, 5])
>>> kser.index.name = 'koalas'
>>> kser.index.name
```

For MultiIndex also

```python
>>> midx = pd.MultiIndex([['lama', 'cow', 'falcon'],
...                       ['speed', 'weight', 'length']],
...                      [[0, 0, 0, 1, 1, 1, 2, 2, 2],
...                       [0, 1, 2, 0, 1, 2, 0, 1, 2]])
>>> pser = pd.Series([45, 200, 1.2, 30, 250, 1.5, 320, 1, 0.3], index=midx)
>>> pser.index.names
FrozenList([None, None])
>>> pser.index.names = ['hello', 'koalas']
>>> pser.index.names
FrozenList(['hello', 'koalas'])

>>> midx = pd.MultiIndex([['lama', 'cow', 'falcon'],
...                       ['speed', 'weight', 'length']],
...                      [[0, 0, 0, 1, 1, 1, 2, 2, 2],
...                       [0, 1, 2, 0, 1, 2, 0, 1, 2]])
>>> kser = ks.Series([45, 200, 1.2, 30, 250, 1.5, 320, 1, 0.3], index=midx)
>>> kser.index.names
[None, None]
>>> kser.index.names = ['hello', 'koalas']
>>> kser.index.names
[None, None]
```

So, this PR suggests that make ours possible also.

```python
>>> kser = ks.Series([1, 2, 3, 4, 5])
>>> kser.index.name = 'koalas'
>>> kser.index.name
'koalas'

>>> midx = pd.MultiIndex([['lama', 'cow', 'falcon'],
...                       ['speed', 'weight', 'length']],
...                      [[0, 0, 0, 1, 1, 1, 2, 2, 2],
...                       [0, 1, 2, 0, 1, 2, 0, 1, 2]])
>>> kser = ks.Series([45, 200, 1.2, 30, 250, 1.5, 320, 1, 0.3], index=midx)
>>> kser.index.names
[None, None]
>>> kser.index.names = ['hello', 'koalas']
>>> kser.index.names
['hello', 'koalas']
```

### CODE REVIEW

File: databricks/koalas/series.py [hunk 1]
```diff
@@ -958,13 +958,11 @@ def rename(self, index: Union[str, Tuple[str, ...]] = None, **kwargs):
     def index(self):
         """The index (axis labels) Column of the Series.
 
-        Currently not supported when the DataFrame has no index.
-
```
Comment (HyukjinKwon) [inline databricks/koalas/series.py]: +1

File: databricks/koalas/series.py [hunk 2]
```diff
@@ -958,13 +958,11 @@ def rename(self, index: Union[str, Tuple[str, ...]] = None, **kwargs):
...
         See Also
         --------
         Index
         """
-        return self.to_frame().index
```
Comment (HyukjinKwon) [inline databricks/koalas/series.py]: Can you just change this to `self._kdf.index`?

Comment (itholic) [reply databricks/koalas/series.py]: @HyukjinKwon ah, yes i can. thanks! :)

File: databricks/koalas/base.py [hunk 3]
```diff
@@ -136,10 +136,13 @@ def _with_new_scol(self, scol: spark.Column) -> IndexOpsMixin
         Creates new object with the new column
     """
     def __init__(self, internal: _InternalFrame, kdf):
+        from databricks.koalas.series import Series
         assert internal is not None
         assert kdf is not None and isinstance(kdf, DataFrame)
         self._internal = internal  # type: _InternalFrame
         self._kdf = kdf
+        if isinstance(self, Series):
+            self._index = kdf.index
```
Comment (HyukjinKwon) [inline databricks/koalas/base.py]: and .. remove the changes here ..

Comment (itholic) [reply databricks/koalas/base.py]: ah, sorry for carelessness.

thanks :)

File: databricks/koalas/base.py [hunk 4]
```diff
@@ -136,6 +136,7 @@ def _with_new_scol(self, scol: spark.Column) -> IndexOpsMixin
         Creates new object with the new column
     """
     def __init__(self, internal: _InternalFrame, kdf):
+        from databricks.koalas.series import Series
```
Comment (HyukjinKwon) [inline databricks/koalas/base.py]: This import too

**Pull request merged**